File: Sources/CountriesFlagsObjC/source/Countries/UIImage+Syria.m
//
//  UIImage+Syria.m
//  CountriesFlags
//

#import "Countries/UIImage+Syria.h"

#import "Extensions/UIBezierPath+Star.h"

@implementation UIImage (Syria)

+ (UIImage *)flagSyrianArabRepublicWithSize:(CGSize)size {
    UIColor* colorRed = [UIColor colorWithRed:206.0f/255.0f
                                        green:17.0f/255.0f
                                         blue:38.0f/255.0f
                                        alpha:1.0f];
    UIColor* colorGreen = [UIColor colorWithRed:0.0f
                                          green:122.0f/255.0f
                                           blue:61.0f/255.0f
                                          alpha:1.0f];
    UIColor* colorWhite = [UIColor whiteColor];
    UIColor* colorBlack = [UIColor blackColor];
    
    UIGraphicsBeginImageContextWithOptions(size, YES, 0.0f);
    
    [colorWhite setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorGreen setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height / 3.0f));
    
    [colorBlack setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 3.0f * 2.0f, size.width, size.height / 3.0f));
    
    [colorRed setFill];
    CGFloat starRadius = size.height / 6.0f * 1.65f / 2.0f;
    [[UIBezierPath starAtPoint:CGPointMake(size.width / 4.0f, (size.height + size.height / 6.0f * 0.15f) / 2.0f)
                        radius:starRadius
                          type:FlagStarTypeFivePointed] fill];
    [[UIBezierPath starAtPoint:CGPointMake(size.width / 2.0f, (size.height + size.height / 6.0f * 0.15f) / 2.0f)
                        radius:starRadius
                          type:FlagStarTypeFivePointed] fill];
    [[UIBezierPath starAtPoint:CGPointMake(size.width * 0.75f, (size.height + size.height / 6.0f * 0.15f) / 2.0f)
                        radius:starRadius
                          type:FlagStarTypeFivePointed] fill];
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image != nil ? image : [UIImage new];
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Syria.swift
//
//  UIImage+Syria.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Syria "wikipedia"
     
     Generates the flag of Syria.
     For more information see [wikipedia].
     
     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Syria.
     */
    internal static func flagSyrianArabRepublic(with size: CGSize) -> UIImage? {
        let colorRed = UIColor(red: 206.0 / 255.0,
                               green: 17.0 / 255.0,
                               blue: 38.0 / 255.0,
                               alpha: 1.0)
        let colorGreen = UIColor(red: 0.0,
                                 green: 122.0 / 255.0,
                                 blue: 61.0 / 255.0,
                                 alpha: 1.0)
        let colorWhite = UIColor.white
        let colorBlack = UIColor.black

        UIGraphicsBeginImageContextWithOptions(size, true, 0.0)

        colorWhite.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorGreen.setFill()
        UIRectFill(CGRect(origin: .zero,
                          size: CGSize(width: size.width, height: size.height / 3.0)))

        colorBlack.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0.0, y: size.height / 3.0 * 2.0),
                          size: CGSize(width: size.width, height: size.height / 3.0)))

        colorRed.setFill()
        let starRadius = size.height / 6.0 * 0.825
        UIBezierPath.star(at: CGPoint(x: size.width / 4.0, y: (size.height + size.height / 6.0 * 0.15) / 2.0),
                          radius: starRadius).fill()
        UIBezierPath.star(at: CGPoint(x: size.width / 2.0, y: (size.height + size.height / 6.0 * 0.15) / 2.0),
                          radius: starRadius).fill()
        UIBezierPath.star(at: CGPoint(x: size.width * 0.75, y: (size.height + size.height / 6.0 * 0.15) / 2.0),
                          radius: starRadius).fill()

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image
    }
}

#endif
